src/parser.rs

```diff
@@ -212,27 +212,36 @@ impl<'a> Parser<'a> {
 
     pub fn path(self) -> Result<'a, String> {
         let (path, parser) = self.word()?;
-        let mut path = Path::new(path);
 
-        // For Linux and BSD and MacOS.
+        let path = Path::new(path);
+        if path.to_str().is_none() {
+            return Ok((String::from(""), parser));
+        }
+
+        let mut path_string = String::from(path.to_str().unwrap());
+
+        // Linux and BSD and MacOS use ~ to infer the home directory of a given user
         if cfg!(unix) && path.starts_with("~") {
             if let Some(base_dirs) = dirs::BaseDirs::new() {
-                let home = base_dirs.home_dir().to_str().unwrap().to_owned();
 
                 if path == Path::new("~") {
-                    return Ok((home, parser));
+                    return Ok((base_dirs.home_dir().to_str().unwrap().to_string(),parser));
                 }
 
-                path = &path.strip_prefix("~").unwrap();
-                return Ok((
-                    Path::new(&home).join(path).to_str().unwrap().to_owned(),
-                    parser,
-                ));
+                let suffix = path.strip_prefix("~").unwrap();
+                path_string = base_dirs.home_dir().join(suffix).to_str().unwrap().to_owned();
```

Comment:
While we're here - I'm a little uncomfortable with the amount of `unwrap()` in 4 lines of code. It would be good to exit this function with `Err` if `to_str()` fails, instead of panicking.

Reply:
I've reduced the amount of unwrapping here in the latest commit, and we already check for the `~` character early on, so I think the chance of an error is negligible.